modules/home-manager/kitty.nix

```diff
@@ -0,0 +1,23 @@
+{ config, lib, ... }:
+let cfg = config.programs.kitty.catppuccin;
+in {
+  options.programs.kitty.catppuccin = with lib; {
+    enable = mkEnableOption "Catppuccin theme";
+    flavour = mkOption {
+      type = types.enum [ "latte" "frappe" "macchiato" "mocha" ];
+      default = config.catppuccin.flavour;
+      description = "Catppuccin flavour for kitty";
+    };
+  };
+
+  config.programs.kitty = with lib;
+    let
+      # string -> string
+      # this capitalizes the first letter in a string
+      # it's used to set the theme name correctly here
+      mkUpper = word:
+        (toUpper (substring 0 1 word)) + (substring 1 (stringLength word) word);
```

Comment:
Can we move this into a `utils.nix` or something? Maybe also the `flavour` and `accent` types.

Reply:
i was thinking of something like that as well
maybe with a `mkCatppuccinOption` function too in order to get rid of the boilerplate from the first part of the files all together?